spa/src/components/MapGL.vue

```diff
@@ -0,0 +1,327 @@
+      return {
+        type: 'FeatureCollection',
+        features: this.vehicles.map((vehicle) => ({
+          type: 'Feature',
+          geometry: {
+            type: 'Point',
+            coordinates: [this.convertLatLng(vehicle.longitude), this.convertLatLng(vehicle.latitude)],
+          },
+          properties: {
+            id: vehicle.id,
+          },
+        })),
+      };
+    },
+    stopsLayer() {
+      return {
+        id: 'stops',
+        type: 'circle',
+        source: 'stops',
+        paint: {
+          'circle-color': [
+            'match',
+            ['get', 'id'],
+            this.focusStop || '',
+            '#1673fc',
+            '#4f96fc',
+          ],
+          'circle-radius': [
+            'match',
+            ['get', 'id'],
+            this.focusStop || '',
+            8,
+            5,
+          ],
+          'circle-stroke-opacity': 0,
+          'circle-stroke-width': 5,
+        },
+      };
+    },
+    vehiclesLayer() {
+      return {
+        id: 'vehicles',
+        type: 'circle',
+        source: 'vehicles',
+        paint: {
+          'circle-color': [
+            'match',
+            ['get', 'id'],
+            this.focusVehicle || '',
+            'red',
+            '#aa2a0f',
+          ],
+          'circle-radius': [
+            'match',
+            ['get', 'id'],
+            this.focusVehicle || '',
+            8,
+            6,
+          ],
+          'circle-stroke-opacity': 0,
+          'circle-stroke-width': 3,
+        },
+      };
+    },
+    focusData() {
+      return (this.focusVehicle && this.vehicles && this.vehicles.find((v) => v.id === this.focusVehicle))
+          || (this.focusStop && this.stops && this.stops.find((s) => s.id === this.focusStop));
+    },
+  },
+  methods: {
+    convertLatLng(value) {
+      return value / 3600000;
+    },
+    onClickMap() {
+      if (!this.focusData) { return; }
+      this.$router.replace({ name: 'map' });
+    },
+    onClickStop(e) {
+      if (this.focusStop === e.mapboxEvent.features[0].properties.id) { return; } // prevent reloading of same stop
+      this.$router.replace({ name: 'mapStop', params: { stop: e.mapboxEvent.features[0].properties.id } });
+    },
+    onClickVehicle(e) {
+      if (this.focusVehicle === e.mapboxEvent.features[0].properties.id) { return; } // prevent reloading of same stop
+      this.$router.replace({ name: 'mapVehicle', params: { vehicle: e.mapboxEvent.features[0].properties.id } });
+    },
+    onMouseEnter(e) {
+      e.map.getCanvas().style.cursor = 'pointer';
+    },
+    onMouseLeave(e) {
+      e.map.getCanvas().style.cursor = '';
+    },
+    onMapLoaded(event) {
+      this.map = event.map;
+    },
+  },
+  created() {
+    // We need to set mapbox-gl library here in order to use it in template
+    this.mapbox = Mapbox;
+    this.map = null;
+  },
+  mounted() {
+    this.$store.dispatch('map/load');
+
+    if (this.focusData) {
```

Comment:
When the map has not been loaded before `this.focusData` will be `null` here even though it should be set.
How could this be fixed? @anbraten

Reply:
The issue should be fixed by https://github.com/anbraten/opnv-live/pull/18/commits/7267e553c6bc8d727d57096e698cd5c359ecf729